drivers/serial/uart_async_to_irq.c

```diff
@@ -259,8 +259,10 @@ void z_uart_async_to_irq_irq_tx_disable(const struct device *dev)
 int z_uart_async_to_irq_irq_tx_ready(const struct device *dev)
 {
 	struct uart_async_to_irq_data *data = get_data(dev);
+	bool ready = (data->flags & A2I_TX_IRQ_ENABLED) && !(data->flags & A2I_TX_BUSY);
 
-	return (data->flags & A2I_TX_IRQ_ENABLED) && !(data->flags & A2I_TX_BUSY);
+	/* async API handles arbitrary sizes */
+	return ready ? INT_MAX : 0;
```

Comment:
`data->tx.len` contains TX buffer size so it should be returned instead of `INT_MAX`.

Reply:
Ah, I thought it was the length of the currently sending buffer, will fix